src/main/java/org/folio/consortia/service/impl/SyncPrimaryAffiliationServiceImpl.java

```diff
@@ -81,38 +68,28 @@ private SyncPrimaryAffiliationBody buildSyncPrimaryAffiliationBody(String tenant
   }
 
   @Override
-  @Async("asyncTaskExecutor")
-  public void createPrimaryUserAffiliations(UUID consortiumId, SyncPrimaryAffiliationBody syncPrimaryAffiliationBody) {
-    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+  public void createPrimaryUserAffiliations(UUID consortiumId, String centralTenantId,
+    SyncPrimaryAffiliationBody syncPrimaryAffiliationBody) {
     log.info("Start creating user primary affiliation for tenant {}", syncPrimaryAffiliationBody.getTenantId());
     var tenantId = syncPrimaryAffiliationBody.getTenantId();
     var userList = syncPrimaryAffiliationBody.getUsers();
-    var centralTenantId = consortiaConfigurationService.getCentralTenantId(tenantId);
```

Comment:
centralTenantId is removed and moved to method parameter because context changes not allowed inside of Async and running method in central tenant context